src/ray/raylet/node_manager.cc

```diff
@@ -451,11 +451,10 @@ void NodeManager::Heartbeat() {
   });
 }
 
-void NodeManager::ReportResourceUsage() {
-  auto resources_data = std::make_shared<rpc::ResourcesData>();
+void NodeManager::FillResourceReport(rpc::ResourcesData *resources_data) {
```

Comment:
Maybe pass by reference? I feel that should be a better practice since it can't be null in this case.